### PR TITLE
[test][sql-gen] Add integration test for table constraints

### PR DESCRIPTION
**Pull request by jaylees14**
- Adds some integration tests for constraints

### CODE REVIEW

File: src/it/scala/temple/generate/database/PostgresGeneratorIntegrationTest.scala [hunk 1]
```diff
@@ -57,6 +60,77 @@ class PostgresGeneratorIntegrationTest extends PostgresSpec with Matchers with B
     result.isLast shouldBe true
   }
 
+  it should "fail when inserting the same data into a column with unique constraint" in {
+    executeWithoutResults(PostgresGenerator.generate(TestData.createStatement))
+    executeWithoutResults(PostgresGenerator.generate(TestData.createStatementWithUniqueConstraint))
+
+    noException should be thrownBy executeWithoutResultsPrepared(
```
Comment (c-harding) [inline src/it/scala/temple/generate/database/PostgresGeneratorIntegrationTest.scala]: `executePreparedWithoutResults` plz

File: src/it/scala/temple/generate/database/PostgresGeneratorIntegrationTest.scala [hunk 2]
```diff
@@ -12,6 +12,9 @@ class PostgresGeneratorIntegrationTest extends PostgresSpec with Matchers with B
 
   // The Postgres container is persisted for every test in this spec: clean up any changes made by each test
   before {
+    executeWithoutResults("DROP TABLE IF EXISTS TestUnique;")
+    executeWithoutResults("DROP TABLE IF EXISTS TestReference;")
+    executeWithoutResults("DROP TABLE IF EXISTS TestCheck;")
```
Comment (c-harding) [inline src/it/scala/temple/generate/database/PostgresGeneratorIntegrationTest.scala]: `CheckTest` et al. plz

**Review by c-harding — DISMISSED** [review]
Couple of suggested renames, but looks good to go

**Review by c-harding — DISMISSED** [review]
more legible, very happy

**Review by Lewiky — APPROVED** [review]

**Review by c-harding — APPROVED** [review]
lgtm (3rd time lucky)

**Pull request merged**